api/loader/resource.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2020 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package loader
+
+import (
+	"context"
+	"log"
+	"os"
+
+	"github.com/yujunz/go-getter"
+	"sigs.k8s.io/kustomize/api/filesys"
+	"sigs.k8s.io/kustomize/api/ifc"
+	"sigs.k8s.io/kustomize/api/internal/git"
+)
+
+type resourceSpec struct {
+	// Raw is the original resource in kustomization.yaml
+	Raw string
+
+	// Dir is where the resource is saved
+	Dir filesys.ConfirmedDir
+}
+
+// Getter is a function that can gets resource
+type resourceGetter func(rs *resourceSpec) error
```

Comment:
Suggest to change `resourceGetter` to `remoteTargetGetter` since `resource` has a specific meaning in kustomize, which is different from the usage here.

Reply:
Done